llvm/lib/Target/X86/X86.td

```diff
@@ -882,7 +886,8 @@ def ProcessorFeatures {
   list<SubtargetFeature> NHMTuning = [TuningMacroFusion,
                                       TuningSlowDivide64,
                                       TuningInsertVZEROUPPER,
-                                      TuningNoDomainDelayMov];
+                                      TuningNoDomainDelayMov,
+                                      TuningAvoidMFENCE];
```

Comment:
Probably add TuningAvoidMFENCE to tuning for x86-64-v2 and later?

Reply:
Also added it to x86-64-v1, since that ought to be equivalent to early "core" marchs?

Reply:
If we do that we might as well support it on any 64-bit capable CPU

Reply:
I believe that's what GCC did.

```
+DEF_TUNE (X86_TUNE_AVOID_MFENCE, "avoid_mfence",
 	 m_CORE_ALL | m_BDVER | m_ZNVER | m_GENERIC)
 ```